Refactor + Extreme toggle

Seirade opened the pull request.

Seirade committed:
Refactor + Extreme toggle
- Removed need for mod_rewrite
- Included jQuery and Bootstrap to cut down on external calls/loading time
- Made Bootstrap CSS/JS version the same
- Replaced icons with suitable emoji equivalents and removed Line Awesome dependency
- Added missing </div> and </html>
- Moved embedded CSS and JS to external files
- Grouped HTML into proper <head> and <body>
- Added missing Bootstrap class to select box
- Added ability to distinguish Extreme games
- Rewrote search results templating to PHP
- Included Extreme check when rendering search results
- Changed array initializer to brackets (requires 5.4+)
- Changed curly brace syntax in strings to simple concatenation, since it's deprecated as of version 7.4 (removed in 8.0)
- Updated README to reflect new requirement of `preferences.json`

puigru commented on common.php:
```diff
@@ -0,0 +1,25 @@
+	}
+	
+	return false;
+}
```
Newline at EOF

puigru commented on common.php:
```diff
@@ -0,0 +1,25 @@
+	}
+}
+
+function is_extreme($game, $nsfw)
```
isExtreme

puigru commented on css/style.css:
```diff
@@ -0,0 +1,17 @@
+{
+	filter: blur(8px);
+	cursor: pointer;
+}
```
Newline at EOF

puigru commented on index.php:
```diff
@@ -1,105 +1,68 @@
+			</div>
+		</div>
+	</body>
+</html>
```
Newline at EOF

puigru commented on js/search.js:
```diff
@@ -0,0 +1,41 @@
+		e.preventDefault();
+	});
+	$('#search .form-control').on('input change', refresh);
+}
```
Newline at EOF

puigru commented on index.php:
```diff
@@ -1,105 +1,68 @@
-<script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
-<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>
-<link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
-<link rel="stylesheet" href="https://maxst.icons8.com/vue-static/landings/line-awesome/line-awesome/1.3.0/css/line-awesome.min.css">
```
Why did you remove line-awesome?

Seirade replied:
It was just another external call/dependency, so it was mainly to cut down on that, but including it just for 3 icons seemed to be overkill.

puigru commented on index.php:
```diff
@@ -1,105 +1,68 @@
+		
+		<title>Flashpoint Browser</title>
+		
+		<script src="js/jquery-3.5.1.min.js"></script>
```
Not convinced by the whole pulling the dependencies into the source tree, but ok.

Seirade replied:
If you want to add dependencies, I would suggest including a `package.json` and then running `npm install` so that they're always local.

puigru replied:
Agreed, although it's out of scope for this PR.

puigru commented on search.php:
```diff
@@ -1,43 +1,46 @@
 <?php
+include 'common.php';
```
require

puigru commented on search.php:
```diff
@@ -1,43 +1,46 @@
+		continue;
+	}
+	
+	echo '<div class="game">';
```
Please use PHP's alternative syntax for control structures when templating HTML.

puigru commented on search.php:
```diff
@@ -1,43 +1,46 @@
+	
+	echo '<div class="game">';
+	echo '<a href="#" data-toggle="collapse" data-target="#game-' . $i . '">[' . $game['platform'] . "] " . $game['title'] . '</a>';
+	echo is_extreme($game, $nsfw) ? ' 🔞' : '';
```
I suggest `<i class="las la-radiation"></i>`

Seirade replied:
Someone's going to obsess over it not matching our `(18)` Extreme icon, and I personally don't like misusing symbols. I settled on just using a Bootstrap pill badge.

puigru replied:
Sure

puigru commented on search.php:
```diff
@@ -1,43 +1,46 @@
+	echo empty($game['activeDataId']) ? '<span title="This game is in LEGACY format."> 📜</span>' : '';
+	echo isset($_POST['fpurl']) ? '<a href="flashpoint://' . $game['id'] . '"> ↪️</a>' : '';
+	echo '<div class="game-details collapse" id="game-' . $i . '" data-id="' . $game['id'] . '">Loading...</div>';
+}
```
Newline at EOF

puigru commented on view.php:
```diff
@@ -1,39 +1,50 @@
-    die('Game not found');
+$blur = is_extreme($game, $nsfw);
+
+if(!$game)
```
Please keep the brace style

Seirade replied:
They specifically throw the following errors in newer versions of PHP:
```
PHP Warning:  Use of undefined constant platform - assumed 'platform' (this will throw an Error in a future version of PHP) in FlashpointBrowser/search.php on line 36
PHP Warning:  Use of undefined constant title - assumed 'title' (this will throw an Error in a future version of PHP) in FlashpointBrowser/search.php on line 36
```
But I added it back in if that's the preference.

puigru replied:
I'm referring to the if statement:
```php
if($game)
{
}
```
vs
```php
if ($game) {
}
```
What you're referring to is a separate issue in search.php

puigru commented on view.php:
```diff
@@ -1,39 +1,50 @@
-        echo "$name: {$game[$key]}\n";
-    }
-}
+	foreach($keys as $key => $name)
```
Ditto

puigru commented on view.php:
```diff
@@ -1,39 +1,50 @@
+	}
 ?>
-</pre>
+</pre>
```
Newline at EOF

puigru commented on view.php:
```diff
@@ -1,39 +1,50 @@
+$baseurl = 'http://infinity.unstable.life/Flashpoint/Data/Images';
+
+$keys =
+[
```
$keys = [

Seirade committed:
Applied recommendations
- Reverted styling changes
- Added npm dependencies
- Updated README

Seirade commented:
Pushed changes, please review.

puigru commented on js/bootstrap.min.js:
```diff
@@ -0,0 +1,7 @@
+/*!
```
You forgot to remove these from the source tree

puigru commented on js/jquery-3.5.1.min.js:
```diff
@@ -0,0 +1,2 @@
+/*! jQuery v3.5.1 | (c) JS Foundation and other contributors | jquery.org/license */
```
This one too

puigru commented on view.php:
```diff
@@ -1,39 +1,46 @@
 <?php
+include 'common.php';
```
Missed this one earlier: require

puigru commented on view.php:
```diff
@@ -1,39 +1,46 @@
 if (!$game) {
-    header('HTTP/1.1 404 Not Found');
-    die('Game not found');
+	header('HTTP/1.1 404 Not Found');
```
Please replace tabs with spaces throughout this file

puigru commented on package.json:
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "flashpointbrowser",
+  "version": "0.0.0",
```
Let's say 0.0.1?

Seirade replied:
Sure, though I was just going to let you update that. Your branch before my changes would be 0.0.1, and after would be 0.0.2?

puigru replied:
I don't really care as we've not been really tracking version so far. 0.0.1 works for me.

Seirade replied:
Ok, I'll just do that then.

puigru commented on .gitignore:
```diff
@@ -0,0 +1,4 @@
+node_modules
+package-lock.json
+preferences.json
+flashpoint.sqlite
```
Newline at EOF

puigru reviewed (changes requested):
Almost there

puigru commented on common.php:
```diff
@@ -0,0 +1,25 @@
+$json = json_decode(file_get_contents('preferences.json'), true);
+$nsfw = [];
+
+foreach($json['tagFilters'] as $filter)
```
Change brace style throughout this file

Seirade committed:
Further changes
- Removed lingering included dependencies
- Changed brace style
- Added newlines

Seirade committed:
Updated version number

puigru committed:
Cleanup whitespace

puigru committed:
CSS style change

puigru committed:
Remove unnecessary newline
